src/harvester/wasser_de.rs

```diff
@@ -0,0 +1,83 @@
+use anyhow::Result;
+use cap_std::fs::Dir;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+use crate::{dataset::Dataset, harvester::Source};
+
+pub async fn harvest(dir: &Dir, client: &Client, source: &Source) -> Result<()> {
+    let url = source
+        .url
+        .join("rest/BaseController/FilterElements/V_REP_BASE_VALID")?;
+
+    let response = client
+        .post(url)
+        .json(&Request { filter: Filter {} })
+        .send()
+        .await?
+        .error_for_status()?
+        .json::<Response>()
+        .await?;
+
+    tracing::info!("Retrieved {} documents", response.results.len().to_string());
+
+    for document in response.results {
+        if let Err(err) = write_dataset(dir, source, document).await {
+            tracing::error!("{:#}", err);
+        }
+    }
+
+    Ok(())
+}
+
+async fn write_dataset(dir: &Dir, source: &Source, document: Document) -> Result<()> {
+    let title = match document.name {
+        Some(name) => name,
+        None => {
+            tracing::warn!("Document {} has no valid entry for 'NAME' ", document.id);
+            return Ok(());
+        }
+    };
+
+    let teaser = document
```

Comment:
Using `unwrap_or(_else)` is certainly nice. The second call could be simplify to `unwrap_or_default` and using [`or_else`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.or_else) might even be bit nicer (but only if you prefer it as well):

```rust
let teaser_text = document.teaser_text.or_else(|| document.auto_teaser_text).unwrap_or_default();
```

Reply:
Actually, just `or` instead of `or_else` should do it:

```rust
let teaser_text = document.teaser_text.or(document.auto_teaser_text).unwrap_or_default();
```